mirrord/layer/src/detour.rs

```diff
@@ -201,6 +205,38 @@ impl<S> Detour<S> {
             Detour::Error(e) => Detour::Error(e),
         }
     }
+
+    /// Return the contained `Success` value or a provided default if `Bypass` or `Error`.
+    ///
+    /// To be used in hooks that are deemed non-essential, and the run should continue even if they
+    /// fail.
+    /// Currently defined only on macos because it is only used in macos-only code.
+    /// Remove the cfg attribute to enable using in other code.
+    #[cfg(target_os = "macos")]
+    pub(crate) fn unwrap_or(self, default: S) -> S {
+        match self {
+            Detour::Success(s) => s,
+            Detour::Bypass(b) => {
+                trace!("Bypassing operation due to {:#?}", b);
+                default
+            }
+            Detour::Error(e) => {
+                trace!("Detour failed with: {:#?}, continuing despite of error.", e);
+                default
+            }
+        }
+    }
+}
+
+/// Allows you to call `.collect()` on an iterator of Detours and get a Detour of a Vec.
+impl<S> FromIterator<Detour<S>> for Detour<Vec<S>> {
+    fn from_iter<T: IntoIterator<Item = Detour<S>>>(iter: T) -> Detour<Vec<S>> {
```

Comment:
Maybe we could implement this with [`Iterator::scan`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.scan) like Rust's [`Result::from_iter`](https://doc.rust-lang.org/src/core/result.rs.html#2068) is saying in a comment?

Reply:
Sounds good, but actually this is not used anywhere yet, I implemented it for something that didn't make it to the final version. I thought if I already have the code I can leave it there in case it's useful for someone in the future. So how about if it's not correct as it is I just delete it and we implement it correctly if we need it in the future?

Reply:
deleted.